tests/sentry/snuba/metrics/test_mqb_query_transformer.py

```diff
@@ -0,0 +1,707 @@
+                ),
+                MetricField(
+                    op="count_web_vitals_measurements",
+                    metric_name="d:transactions/measurements.fid@millisecond",
+                    args=["meh"],
+                    alias="count_web_vitals_measurements_fid_meh",
+                ),
+                MetricField(
+                    op="count_web_vitals_measurements",
+                    metric_name="d:transactions/measurements.fp@millisecond",
+                    args=["good"],
+                    alias="count_web_vitals_measurements_fp_good",
+                ),
+                MetricField(
+                    op="count_web_vitals_measurements",
+                    metric_name="d:transactions/measurements.lcp@millisecond",
+                    args=["good"],
+                    alias="count_web_vitals_measurements_lcp_good",
+                ),
+                MetricField(
+                    op="count_web_vitals_measurements",
+                    metric_name="d:transactions/measurements.fcp@millisecond",
+                    args=["meh"],
+                    alias="count_web_vitals_measurements_fcp_meh",
+                ),
+            ],
+            start=datetime.datetime(2022, 3, 24, 11, 11, 35, 447729, tzinfo=pytz.utc),
+            end=datetime.datetime(2022, 6, 22, 11, 11, 35, 447729, tzinfo=pytz.utc),
+            granularity=Granularity(granularity=3600),
+            where=None,
+            groupby=[MetricGroupByField("transaction", alias=None)],
+            limit=Limit(limit=51),
+            offset=Offset(offset=0),
+            include_series=False,
+        ),
+        id="count_web_vitals test case",
+    ),
+    pytest.param(
+        Query(
+            match=Entity("generic_metrics_distributions"),
+            select=[
+                Function(
+                    function="p95",
+                    initializers=None,
+                    parameters=[Column("d:transactions/duration@millisecond")],
+                    alias="p95",
+                ),
+                Function(
+                    function="rate",
+                    initializers=None,
+                    parameters=[
+                        Column("d:transactions/duration@millisecond"),
+                        60,
+                        7776000.0,
+                    ],
+                    alias="epm",
+                ),
+                Function(
+                    function="e:transaction/failure_rate@ratio",
+                    initializers=None,
+                    parameters=[],
+                    alias="failure_rate",
+                ),
+            ],
+            groupby=[
+                AliasedExpression(
+                    exp=Column("transaction.status"),
+                    alias="transaction.status",
+                ),
+                Function(
+                    function="e:transactions/team_key_transactions@none",
```

Comment:
so this is basically a derived op like histogram and rate, while others are just derived names and so they have an MRI while this doesn't

Reply:
Right! I guess you could also model `team_key_transactions` as an op though, right? Not saying that would be better.

Reply:
Yeah you are right. I thought it might make more sense to reserve derived ops for functions that are applied on a metric, and team_key_transactions sort of logically falls into the derived metric vicinity by that reasoning

Reply:
Can we document the difference between derived ops vs functions somewhere? Its still not quite intuitive to me what the difference is

Reply:
Should be clarified here https://github.com/getsentry/sentry/pull/37652/files#diff-cb864c73ed0f0c9274f8f1bced86b30f7b5708d28befdef6d7c8e1353cf08ef6R18